### ISSUE TITLE
testing function fails when one of the args is a user defined function

### ISSUE THREAD

**Issue by machow** [datacamp/pythonwhat#155]
Example below. Note that if you replace `sum2` with the built-in `sum`, then it is fine. @hugobowne should be a quick fix, will submit pull request by EOD Friday or let you know.

*** =solution

```{python}
def sum2(arr): return sum(arr)

def apply(f, arr):
    return f(arr)

apply(sum2, [1,2,3])
```

*** = sct
```{python}
test_function('apply')
```

Error:  `AttributeError: Can't get attribute 'sum2' on <module '__main__' from 'tests/test_test_function.py'>`

**Comment by machow** [comment]
### tl;dr - use test_function_v2 w/ do_eval to False for the user defined function.

*** =sct
```{python}
test_function_v2('apply', params=['f', 'arr'], do_eval=[False, True])
```

### explanation

This issue is a bit more complicated. In the SCT above `test_function('apply')` will need to verify that `sum2` is the same for the submission and solution code. In this case using `do_eval=true` doesn't make sense, because both the student and submission define the function separately, so even if we could grab the functions for `sum2` from both environments they wouldn't be identical.

In other words

```{python}
# solution code
def f(a): print(a)

f1 = f

# submission code
def f(a): print(a)

f2 = f

# what pythonwhat would be doing
f1 == f2   # False
```

(CC @hugobowne)

**Comment by machow** [comment]
in production, example above was added as unit test in test_test_function.py, now fails gracefully rather than raising error.
